### PR TITLE
Small refactoring + error reporting

### PR DESCRIPTION
**Pull request by MoOx**
Just a PR following my stupid
https://github.com/reworkcss/rework-calc/pull/3

I’ve refactored the code a little bit (& also use rework-visit which is
more bullet proof) like rework-vars.
I quickly refactor tests to make it more understandable.
I add some error reporting like we have in rework-vars.

Btw, maybe we can expand the use of balanced-match to rewrite other
parts of the plugin.

poke @necolas

### CODE REVIEW

File: lib/calc.js [hunk 1]
```diff
@@ -1,24 +1,20 @@
-
 /**
```
Comment (necolas) [inline lib/calc.js:1]: let's just remove this comment, doesn't belong in the source code

File: lib/calc.js [hunk 2]
```diff
@@ -28,58 +24,47 @@
...
- * Visit all declarations (in a rule)
- *
- * @param {Array} declarations
- * @api private
- */
-function visit(declarations) {
-    declarations.forEach(function (decl) {
-        if (!hasExpressions(decl.value)) return;
+module.exports = function calc(style) {
+  // resolve variables
```
Comment (necolas) [inline lib/calc.js]: variables?

File: lib/calc.js [hunk 3]
```diff
@@ -28,58 +24,47 @@
...
+    for (var i = 0; i < declarations.length; i++) {
+      decl = declarations[i];
+      value = decl.value;
 
-        evaluateAndApplyExpressions(expressions, decl);
-    });
-}
+      // skip comments
+      if (decl.type !== 'declaration') continue;
+      // skip values that don't contain variable functions
```
Comment (necolas) [inline lib/calc.js]: calc functions

File: lib/calc.js [hunk 4]
```diff
@@ -118,19 +103,24 @@ function getExpressionsFromValue(value) {
...
-        // Insert the evaluated value:
-        var expRegexp = new RegExp(
-                                EXPRESSION_METHOD_REGEXP + '\\(' +
-                                escapeExp(expression) + '\\)'
-                            );
-        declaration.value = declaration.value.replace(expRegexp, result);
-    });
+function resolveValue(value) {
+  var balancedParens = balanced('(', ')', value);
+  var calcRef = balanced(CALC_FUNC_IDENTIFIER + '(', ')', value);
```
Comment (necolas) [inline lib/calc.js]: this is buggy. see latest commits to rework-vars for how i fixed that module

File: lib/calc.js (hunk 2 above)
Comment (MoOx) [inline lib/calc.js]: copy pasted :(

**Comment by MoOx** [comment]
@necolas, thanks for the review, I'll check rework-vars updates & handle your comments tomorrow morning + release 0.3.0
Or maybe it's time to ship 1.0.0.

**Pull request merged**